src/SecureHeaders.php

```diff
@@ -1246,30 +1189,32 @@ private function deconstructHeaderValue(
         return $attributes;
     }
 
-    private function validateHeaders()
+    private function validateHeaders(HeaderBag $headers)
```

Comment:
The actual attribute decomposition isn't doing what you expect (i.e. the later calls to `$header->hasAttribute('report-uri')` etc.. won't work because even when `report-uri` exists in the CSP, it isn't being pulled out in the way that the code expects)